From 08451999e59c58d8f6c8e3de57f7750f535832f6 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Thu, 9 Aug 2018 19:20:37 +0800
Subject: [PATCH 4413/5242] MLK-19179: clk: imx8mm: change audio ahb and ipg
 clock to 400M

commit  7fbd6e9a2e2e552491426cdac345d69573d9b34a from
https://source.codeaurora.org/external/imx/linux-imx.git

According to ADD, the audio ahb and ipg clock should be in 1:1 mode
and the frequency is 400MHz

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
(cherry picked from commit ee175a8cea1a7d27954a73c3447bb16edd71f4c8)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx8mm.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/imx/clk-imx8mm.c b/drivers/clk/imx/clk-imx8mm.c
index e4e0c80..4381f57 100644
--- a/drivers/clk/imx/clk-imx8mm.c
+++ b/drivers/clk/imx/clk-imx8mm.c
@@ -956,7 +956,9 @@ static void __init imx8mm_clocks_init(struct device_node *ccm_node)
 			clk_prepare_enable(clks[clks_init_on[i]]);
 	}
 
-	clk_set_parent(clks[IMX8MM_CLK_AUDIO_AHB_SRC], clks[IMX8MM_SYS_PLL2_500M]);
+	clk_set_parent(clks[IMX8MM_CLK_AUDIO_AHB_SRC], clks[IMX8MM_SYS_PLL1_800M]);
+	clk_set_rate(clks[IMX8MM_CLK_AUDIO_AHB_DIV], 400000000);
+	clk_set_rate(clks[IMX8MM_CLK_IPG_AUDIO_ROOT], 400000000);
 
 	/* increase NOC clock to design target */
 	clk_set_rate(clks[IMX8MM_SYS_PLL3], 750000000);
-- 
1.7.9.5

